packages/instant/src/util/gas_price_estimator.ts

```diff
@@ -0,0 +1,44 @@
+import { BigNumber } from '@0x/utils';
+
+import { DEFAULT_GAS_PRICE } from '../constants';
+
+interface GasStationResult {
+    average: number;
+    fastestWait: number;
+    fastWait: number;
+    fast: number;
+    safeLowWait: number;
+    blockNum: number;
+    avgWait: number;
+    block_time: number;
+    speed: number;
+    fastest: number;
+    safeLow: number;
+}
+
+const endpointUrl = 'https://ethgasstation.info/json/ethgasAPI.json';
+const fetchFastAmountInWei = async () => {
+    const res = await fetch(endpointUrl);
+    const gasInfo = (await res.json()) as GasStationResult;
+    const gasPriceInGwei = new BigNumber(gasInfo.fast / 10);
+    return gasPriceInGwei.mul(1000000000);
+};
+
+export class GasPriceEstimator {
+    private _lastFetched?: BigNumber;
```

Comment:
Should we have a way of invalidating this state? Ie. a concept of staleness?

Reply:
I think it's okay to be naive in this scenario.  The flow is:

```
- On startup, get best gas price
- When submitting txn, try to fetch fresh price, but if fails, fall back to last known price or resort to default
```

I don't think staleness is necessary in the scenario, because I would expect the last fetched price to be more accurate than falling back to the default. Open to feedback though